src/open_inwoner/cms/cases/views/services.py

```diff
@@ -112,26 +115,36 @@ def get_submissions_for_api_group(
 
     def get_submissions(self) -> list[SubmissionWithApiGroup]:
         all_api_groups = list(
-            ZGWApiGroupConfig.objects.exclude(form_service__isnull=True)
+            ZGWApiGroupConfig.objects.select_related(
+                "form_service",
+            ).exclude(form_service__isnull=True)
         )
 
-        with parallel() as executor:
+        subs_with_api_group: list[SubmissionWithApiGroup] = []
+        with parallel(max_workers=self._max_workers) as executor:
             futures = [
                 executor.submit(self.get_submissions_for_api_group, group)
                 for group in all_api_groups
             ]
 
-            subs_with_api_group: list[SubmissionWithApiGroup] = []
-            for task in concurrent.futures.as_completed(futures):
-                try:
-                    group_for_task = all_api_groups[futures.index(task)]
-                    subs_with_api_group.extend(
-                        SubmissionWithApiGroup(submission=row, api_group=group_for_task)
-                        for row in task.result()
-                    )
+            try:
+                for task in concurrent.futures.as_completed(
+                    futures,
+                    timeout=self._timeouts["fetch_submissions"],
+                ):
+                    try:
+                        group_for_task = all_api_groups[futures.index(task)]
+                        subs_with_api_group.extend(
+                            SubmissionWithApiGroup(
+                                submission=row, api_group=group_for_task
+                            )
+                            for row in task.result()
+                        )
 
-                except BaseException:
-                    logger.exception("Error fetching and pre-processing cases")
+                    except BaseException:
+                        logger.exception("Error fetching and pre-processing cases")
+            except TimeoutError:
```

Comment:
Are you using the builtin `TimeoutError`? I thought that one is for OS-related errors (e.g. the opening of a file taking too long) rather than I/O bound operations. There is a `concurrent.futures.TimeoutError`. Why not use this?